MM-41988 Updated links to legacy domain about.mm.com

cwarnermm opened the pull request:
#### Summary
Updated in-product links to legacy domain about.mattermost.com.

#### Release Note

```release-note
Updated in-product links to legacy domain about.mattermost.com
```

saturninoabril commented:
/update-branch

saturninoabril commented:
/update-branch

saturninoabril reviewed (changes requested):
Thanks Carrie! Could you please check other similar links as a result when searching for `about.mattermost.com`?

cwarnermm commented:
@saturninoabril Thanks! I believe I've updated the remaining legacy URLs.

isacikgoz reviewed (approved):
LGTM 👍

saturninoabril commented on i18n/bg.json:
```diff
@@ -2473,7 +2473,7 @@
   {
     "id": "app.system.warn_metric.bot_description",
-    "translation": "[Научете повече за Mattermost съветника](https://about.mattermost.com/default-channel-handle-documentation)"
+    "translation": "[Научете повече за Mattermost съветника](https://docs.mattermost.com/manage/in-product-notices.html#admin-advisor-notices)"
```
(non-blocking for me) I believe we are only changing translation strings at `en.json` and Translation team will follow through the necessary changes per language owner. So, I don't know if this could be an exception.

saturninoabril reviewed (approved):
Thanks Carrie! Left one comment but non-blocking for me. Feel free to merge.

hmhealey commented on model/config.go:
```diff
@@ -128,9 +128,9 @@ const (
-	SupportSettingsDefaultAboutLink          = "https://about.mattermost.com/default-about/"
-	SupportSettingsDefaultHelpLink           = "https://about.mattermost.com/default-help/"
-	SupportSettingsDefaultReportAProblemLink = "https://about.mattermost.com/default-report-a-problem/"
+	SupportSettingsDefaultAboutLink          = "https://docs.mattermost.com/about/product.html/"
```
@cwarnermm Like I mentioned in the web app PR, we need to keep these old redirects around since almost all existing Mattermost instances will be pointing to them.

I still think it's a good idea to have them pointing to redirects as well for a similar reason, but that's not as necessary

cwarnermm commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -1,12 +1,12 @@
 
-For troubleshooting see: http://forum.mattermost.org/.
-For feature proposals see: http://www.mattermost.org/feature-requests/
+For troubleshooting see: https://forum.mattermost.com//.
```
```suggestion
For troubleshooting see: https://forum.mattermost.com/.
```

cwarnermm commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -1,12 +1,12 @@
-For troubleshooting see: http://forum.mattermost.org/.
-For feature proposals see: http://www.mattermost.org/feature-requests/
+For troubleshooting see: https://forum.mattermost.com//.
+For feature proposals see: https://www.mattermost.org/feature-ideas/
```
```suggestion
For feature proposals see: https://www.mattermost.com/feature-ideas/
```

cwarnermm commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -1,12 +1,12 @@
-For troubleshooting see: http://forum.mattermost.org/.
-For feature proposals see: http://www.mattermost.org/feature-requests/
+For troubleshooting see: https://forum.mattermost.com/.
+For feature proposals see: https://www.mattermost.org/feature-ideas/
```
```suggestion
For feature proposals see: https://www.mattermost.com/feature-ideas/
```

cwarnermm commented:
@hmhealey & @saturninoabril - I believe that all paused/outstanding work on this PR has now been completed. I have removed all of the language files I touched with URL updates (except the en.json file), and I replaced the about.mm.com legacy URLs you flagged as potentially problematic with the new destination links Emily made available. Removing the "Do not merge" label for final reviews.

cwarnermm commented:
/update-branch

hmhealey commented:
Sorry, by removing the other language files, I meant removing the changes to them and reverting to the existing versions which is why these are all conflicting now. Let me know if you need help doing that since it looks like you're using the web interface to modify files, and I'm not sure if there's an easy way to do that from here

cwarnermm commented:
@hmhealey - I identified and fixed a couple of strings. I believe we're good to go now.

saturninoabril commented:
/update-branch

saturninoabril commented:
@cwarnermm Build failed due to failed tests. I'll defer to @isacikgoz if you need help in sorting out those tests.

saturninoabril reviewed (changes requested):
Looks good to me, just need to address failed tests. Thanks Carrie for updating!

cwarnermm commented:
@isacikgoz - Would you be able to assist with the build failures, please?

saturninoabril reviewed (approved):
Thanks @cwarnermm and @isacikgoz! LGTM.

saturninoabril commented on i18n/en.json:
```diff
@@ -1016,7 +1016,7 @@
   {
     "id": "api.command_mute.no_channel.error",
-    "translation": "Could not find the specified channel. Please use the [channel handle](https://about.mattermost.com/default-channel-handle-documentation) to identify channels."
+    "translation": "Could not find the specified channel. Please use the [channel handle](https://docs.mattermost.com/messaging/managing-channels.html#naming-a-channel) to identify channels."
```
@cwarnermm @emdecr I noticed while reviewing webapp changes that this redirects to `https://docs.mattermost.com/guides/channels.html#work-with-channels`. Should that be used here instead?

cwarnermm commented:
@saturninoabril - Thanks for calling this out! Yes, this URL should likely be updated. The good news is that we have a redirect in place to catch this in the meantime and take the user to the correct page in the docs. Would you possibly be open to updating that link? Otherwise, please log this issue and assign it to me.

saturninoabril commented:
Hello @cwarnermm, you're welcome and sure, I'll take care of it since it will require a follow up change to webapp E2E as well. And no worries 😉.

Great job by the way!
